src/main/resources/dlang/ide/liveTemplates/D.xml

```diff
@@ -0,0 +1,469 @@
+      <variable name="NAME" expression="" defaultValue="&quot;myfunction&quot;" alwaysStopAt="true" />
+      <variable name="PARAMETER1TYPE" expression="" defaultValue="&quot;int&quot;" alwaysStopAt="true" />
+      <variable name="PARAMETER1NAME" expression="" defaultValue="&quot;x&quot;" alwaysStopAt="true" />
+      <variable name="DPARAMETER1NAME" expression="PARAMETER1NAME" defaultValue="" alwaysStopAt="false" />
+      <variable name="VOID" expression="" defaultValue="&quot;void&quot;" alwaysStopAt="true" />
+      <variable name="EX" expression="" defaultValue="&quot;@safe pure nothrow @nogc&quot;" alwaysStopAt="true" />
+      <variable name="BRIEFSUMMARY" expression="" defaultValue="&quot;Brief summary of what&quot;" alwaysStopAt="true" />
+      <variable name="DNAME" expression="NAME" defaultValue="" alwaysStopAt="false" />
+      <variable name="BRIEFSUMMARY2" expression="" defaultValue="&quot;does, forming the summary section&quot;" alwaysStopAt="true" />
+      <variable name="Description" expression="" defaultValue="&quot;First paragraph of synopsis description.&quot;" alwaysStopAt="true" />
+      <variable name="DVOID" expression="VOID" defaultValue="" alwaysStopAt="true" />
+      <context>
+        <option name="D" value="true" />
+      </context>
+    </template>
+    <template name="this" value="/// Description &#10;this($PARAMETER1TYPE$ $PARAMETER1NAME$) $EX$&#10;{ &#10;    $END$&#10;}&#10;" description="constructor (this)" toReformat="false" toShortenFQNames="true">
+      <variable name="PARAMETER1TYPE" expression="" defaultValue="&quot;int&quot;" alwaysStopAt="true" />
+      <variable name="PARAMETER1NAME" expression="" defaultValue="&quot;x&quot;" alwaysStopAt="true" />
+      <variable name="EX" expression="" defaultValue="&quot;@safe pure nothrow @nogc&quot;" alwaysStopAt="true" />
+      <context>
+        <option name="D" value="true" />
+      </context>
+    </template>
+    <template name="get" value="$TYPE1$ $NAME2$() $ACCESS$ &#10;{ &#10; return $FIELD$; &#10;} &#10;" description="getter property (get)" toReformat="false" toShortenFQNames="true">
+      <variable name="TYPE1" expression="" defaultValue="&quot;int&quot;" alwaysStopAt="true" />
+      <variable name="NAME2" expression="" defaultValue="&quot;getter&quot;" alwaysStopAt="true" />
+      <variable name="ACCESS" expression="" defaultValue="&quot;@safe pure nothrow const @nogc&quot;" alwaysStopAt="true" />
+      <variable name="FIELD" expression="concat(NAME2, &quot;_&quot;)" defaultValue="" alwaysStopAt="true" />
+      <context>
+        <option name="D" value="true" />
+      </context>
+    </template>
+    <template name="set" value="void $NAME1$($TYPE2$ rhs) ${3:@safe pure nothrow @nogc} &#10;{ &#10; ${0:$1_} = rhs; &#10;} &#10;" description="setter property (set)" toReformat="false" toShortenFQNames="true">
```

Comment:
When I used this while testing(on 2019 ide), I got:
```
void Bar(int rhs) ${3:@safe pure nothrow @nogc}
	{ 
	 ${0:$1_} = rhs; 
	}
```
Not sure exactly why, though.

Reply:
Also seeing similar behavior on some but not all Live Templates.

Reply:
Only the goddess of D should write template definitions. Unfortunately I began with it:)
These unix-shell like variable definitions are remnants from when i searched around choosing some basic templates to begin with. I just wanted something working and thought they evolve as (hopefully) my skills in D evolve.
Thankfully you are D gods:)
I've run through all the templates in da1ece7 and fixed the variable names.

Reply:
Side-note: This is really a XML-Hell. 
Despite that this is horrible to look at as a diff (git or github view), when you edit existing templates and export them, a diff-XML (this time from IDEA) is produced and i had to manually insert every single template back into the master file of the plugin. I managed to make 3 mistakes and forgot one template. But I was not alone. CLion forgot the _context_ definition of my rewritten templates:)